### ISSUE TITLE
improvement: custom labels on BarChart

### ISSUE THREAD

**Issue by ADevelopStudio** [danielgindi/Charts#1095]
if in file HorizontalBarChartRenderer.swift  (line 310) change : 
`                        let valueText = formatter.stringFromNumber(val)!
` 
to:
```     
 var valueText = formatter.stringFromNumber(val)!     
 if let labelValue = e.data as? String {
        valueText = labelValue
 }
``` 

and add this label to 
``` 
 BarChartDataEntry(value: values[i], xIndex: i, data: "YOUR_CUSTOM_LABEL_AS_STRING")
``` 

it will add custom labels to chart 
![simulator screen shot 31 may 2016 21 56 49](https://cloud.githubusercontent.com/assets/5238454/15673519/97c0dec6-277a-11e6-9066-884fa27a4e62.png)

**Comment by liuxuan30** [comment]
It's a little obscure to figure out the difference, what you want to achieve? I guess value formatter can take care of the prefix and appendix

**Comment by ADevelopStudio** [comment]
this graph displays daily value of some vitamins, as an example. 
so bar represents percentage but labels on the right shows exact amount. 
because not app vitamins (in my case) has the same daily value. I mean some of them can be 1g and some 0.01g. As you can see on the chart, Vitamin K has only 154 mcg ( 0.154mg)  but daily percentage  is much bigger than Vitamin E with 2mg. 
You can't do it with value formatter, first of all because  prefix and appendix are common for all values. you can't use different for each bar. Plus here I display the exact amount as label and daily percentage as bar at the same time

**Comment by liuxuan30** [comment]
I see. For now, I would suggest you do some customizations to meet your goal.
@danielgindi seems more and more people are asking about custom value labels

**Comment by ADevelopStudio** [comment]
no worries, I reached my goal by doing this way. I just wanted to share my suggestion

**Comment by hardikdevios** [comment]
+1

**Comment by danielgindi** [comment]
Yes, and there are many open issues... 
I need some help here managing the issues, marking duplicates/bugs/feature requests etc.
I have more work now taking care of a new baby girl :-)

**Comment by liuxuan30** [comment]
I am trying to as much as possible to label them. BTW, congrats

**Comment by danielgindi** [comment]
Implemented in Charts 3.0 (v3 branch)
